sql/core/src/test/scala/org/apache/spark/sql/execution/arrow/ArrowConvertersSuite.scala

```diff
@@ -1629,6 +1632,39 @@ class ArrowConvertersSuite extends SharedSQLContext with BeforeAndAfterAll {
     }
   }
 
+  test("roundtrip payloads") {
+    val allocator = ArrowUtils.rootAllocator.newChildAllocator("int", 0, Long.MaxValue)
+    val vector = ArrowUtils.toArrowField("int", IntegerType, nullable = true)
+      .createVector(allocator).asInstanceOf[NullableIntVector]
+    vector.allocateNew()
+    val mutator = vector.getMutator()
+
+    (0 until 10).foreach { i =>
+      mutator.setSafe(i, i)
+    }
+    mutator.setNull(10)
+    mutator.setValueCount(11)
+
+    val schema = StructType(Seq(StructField("int", IntegerType)))
+
+    val batch = new ColumnarBatch(schema, Array[ColumnVector](new ArrowColumnVector(vector)), 11)
```

Comment:
Btw, do we need to use `ColumnarBatch` for this test?
I guess we can simply create `Iterator[InternalRow]` and use it.

Reply:
you mean just calling something like `new ColumnarBatch(..).rowIterator()`?  We still need to set the number of rows in the batch I believe

Reply:
Oh, you mean not using `ArrowColumnVector` at all and just make an `Iterator[InternalRow]` some other way?  That would probably work, but I figured why not test out the columnar batch this way also.

Reply:
Yes, I meant your second comment.
We do test the columnar batch with `ArrowColumnVector` in `ColumnarBatchSuite` and also we use it in `ArrowConverters.fromPayloadIterator()`, so I thought we don't need to use it here.